src/Control/Bind.purs

```diff
@@ -63,6 +63,17 @@ infixr 1 bindFlipped as =<<
 instance bindFn :: Bind ((->) r) where
   bind m f x = f (m x) x
 
+-- | The `Array` monad's "do notation" works like a nested for loop. Each
+-- | `bind`/`elem <- array` adds another level of nesting in the loop:
+-- | ```
+-- | foo :: Array Int
+-- | foo = do
+-- |   eachElementInArray1 <- [0, 1]
+-- |   eachElementInArray2 <- [1, 2]
+-- |   pure (eachElementInArray1 + eachElementInArray2)
```

Comment:
I think this may be clearer if we use a different operation here. How about replacing `[0,1]` with `["A", "B"]` and `[1,2]` with `["C","D"]`, and then combining them with `<>` instead of `+`? That way each element of the result array is distinct, which means that it's easier to see the order in which the elements are processed.

Reply:
Agreed! Updated in latest commit.